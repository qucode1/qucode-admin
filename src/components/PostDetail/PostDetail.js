import React from 'react'
import Post from '../Post/Post'
import NotFound from '../NotFound/NotFound'
import { Route } from 'react-router-dom'

const PostDetail = function (props) {
  const matchingPost = props.posts.posts.find((post) => post.slug === props.match.params.slug)
  return (
    <div>
      {matchingPost
        ? <Post {...matchingPost} details />
        : <Route component={() => <NotFound message='Post not found' />} />}
    </div>
  )
}

export default PostDetail
