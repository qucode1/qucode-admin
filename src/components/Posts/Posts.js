import React from 'react'
import { Link } from 'react-router-dom'

import Post from '../Post/Post'
import styles from '../Post/Post.css'

const Posts = (props) => (
  <div className='main'>
    {props.posts.posts.map((post) => (
      <Post key={post.slug} {...post} details={false} />
    ))}
    <ul>
      <li>
        <Link to='/posts/test'><h2 className={styles.title}>Using TravisCI (will not work)</h2></Link>
      </li>
      <li>
        <Link to='/test'>Go Somewhere (Won't work)</Link>
      </li>
      <li>
        <Link to='/about/cyxc/7sy'>Go Somewhere (Won't work)</Link>
      </li>
    </ul>
    <style jsx>{`
    `}</style>
  </div>
)

export default Posts
