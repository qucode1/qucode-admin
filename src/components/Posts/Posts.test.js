import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

// import posts from '../../blog-posts.json'
import Posts from './Posts'
import PostDetail from '../PostDetail/PostDetail'

const posts = {
  'posts': [{
    'slug': 'post-slug'
  }]
}

const match = {
  'url': '/posts',
  'params': {
    'slug': 'post-slug'
  }
}

test('Should render Post List', () => {
  const component = mount(
    <MemoryRouter initialEntries={['/posts']} initialIndex={0} >
      <Posts posts={posts}/>
    </MemoryRouter>
  )

  expect(component.find('ul').length).toBe(1)
})

it('Posts Component should render as expected', () => {
  const wrapper = shallow(<Posts posts={posts}u/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})
