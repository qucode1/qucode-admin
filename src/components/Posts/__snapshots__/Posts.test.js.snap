// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component should render as expected 1`] = `
<div>
  <Post
    details={false}
    slug="post-slug"
  />
  <ul>
    <li>
      <Link
        replace={false}
        to="/posts/test"
      >
        <h2
          className="title"
        >
          Using TravisCI (will not work)
        </h2>
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/test"
      >
        Go Somewhere (Won't work)
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/about/cyxc/7sy"
      >
        Go Somewhere (Won't work)
      </Link>
    </li>
  </ul>
</div>
`;
